### DISCUSSION TITLE
Restore db to an older schema

### DISCUSSION THREAD

**Discussion by niamR** [stashapp/stash#3688] [Q&A] [answered]
Hi there,

Updating to 0.20.2 (not sure from which source version) I had to perform schema upgrades from 43 to 45.
It ended up with a corrupted database - thought I didn't noticed any failure during the schema upgrade process from browser page, didn't looked at the logs at that time thought.

So now I would like to do a db restore from the existing backup took right before the schema upgrade.
Following the wiki I replaced the current stash-go.sqlite by my schema 43 file.

Now when I restart my container it gives the following error:

stash  | INFO[2023-04-26 22:56:28] using config file: /config/config.yml        
stash  | panic: schema version 45 is incompatible with required schema version 43 [recovered]
stash  |        panic: Panic: schema version 45 is incompatible with required schema version 43
stash  | 
stash  | goroutine 1 [running]:
stash  | main.handleError({0x1af65e0?, 0x400009e000})
stash  |        github.com/stashapp/stash/cmd/stash/main.go:53 +0x80
stash  | main.recoverPanic()
stash  |        github.com/stashapp/stash/cmd/stash/main.go:44 +0x54
stash  | panic({0x1397b20, 0x4000040460})
stash  |        runtime/panic.go:884 +0x20c
stash  | main.main()
stash  |        github.com/stashapp/stash/cmd/stash/main.go:24 +0x50

I didn't find anywhere to change the schema version "manually" in the config files.
Any idea ?

**Comment by DogmaDragon** [comment, marked as answer]
If you want to restore the older working version, make sure to run older executable. 

If you want to run migration again from the backup, go to `localhost:9999/migrate` if it doesn't start automatically.

**Comment by Flashy78** [comment]
Your error indicates your database is still schema 45 even though you downgraded Stash to 43.
